src/controller/deletecontent.js

```diff
@@ -99,8 +123,23 @@ function shouldAutoparagraph( doc, position ) {
 	return !isTextAllowed && isParagraphAllowed;
 }
 
-function checkCanBeMerged( element ) {
-	const schema = element.document.schema;
+function checkCanBeMerged( left, right ) {
+	const schema = left.document.schema;
```

Comment:
It's nice and all, but couldn't you just use `Treewalker` and see what objects you are stepping over?